### PR TITLE
Removed hardcoded shapes from LogSoftmax operation

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Removed branch which has created the Reshape with hardcoded shapes. After this branch we hadn't be able to make a reshape to another shape for the nGraph function.

### Tickets:
 - 45000

### CODE REVIEW

**Comment by ilyachur** [comment]
@tomdol I didn't find a place to add new tests for this change. Do we have tests which check that we can reshape nGraph function after the ONNX importer?

**Review by tomdol — APPROVED** [review]

**Comment by tomdol** [comment]
This is tested in e2e tests but not per-op, it's done per-model and I'm not sure if LogSoftmax is used in any of them. Let me check the code and get back to you about it.

**Comment by ilyachur** [comment]
@tomdol  I added new tests. Please have a look.

**Review by jane-intel — APPROVED** [review]

File: ngraph/test/onnx/onnx_import.in.cpp [hunk 1]
```diff
@@ -3546,6 +3547,18 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_logsoftmax13_2D)
...
 
+NGRAPH_TEST(${BACKEND_NAME}, onnx_model_logsoftmax13_2D_reshape)
+{
+    const auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/logsoftmax13_2D.prototxt"));
+    InferenceEngine::CNNNetwork net(function);
+    InferenceEngine::ICNNNetwork::InputShapes shapes = {};
+    InferenceEngine::SizeVector shape = {1, 1, 4000};
+    shapes[net.getInputsInfo().begin()->first] = shape;
+    net.reshape(shapes);
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import.in.cpp:3558]: Optionally you could wrap this line in a EXPECT_NO_THROW() macro but alltogether looks good.

Comment (ilyachur) [reply ngraph/test/onnx/onnx_import.in.cpp]: Sure, fixed

**Pull request merged**